pkg/ccip/chainaccessor/ton_accessor.go

```diff
@@ -181,13 +188,136 @@ func (a *TONAccessor) Sync(ctx context.Context, contractName string, contractAdd
+	// get onramp address
+	onrampAddr, err := a.getBinding(consts.ContractNameOnRamp)
+	if err != nil {
+		return nil, fmt.Errorf("OnRamp not bound: %w", err)
+	}
+
+	// 	CCIPMessageSent {
+	//   Message: TVM2AnyRampMessage (in reference cell) {
+	//     Header: RampMessageHeader {
+	//       MessageID: 256 bits
+	//       SourceChainSelector: 64 bits
+	//       DestChainSelector: 64 bits    <- offset: 40
+	//       SequenceNumber: 64 bits       <- offset: 48
+	//       Nonce: 64 bits
+	//     }
+	//     Sender: address
+	//     Body: TVM2AnyRampMessageBody (in reference)
+	//     FeeValueJuels: 96 bits
+	//   }
+	// }
+
+	// query TON logs
+	res, err := logpoller.NewQuery[onramp.CCIPMessageSent](onrampAddr, hash.CRC32("CCIPMessageSent")).
+		SkipBytes(40). // Skip to DestChainSelector
+		FilterBytes(8, query.EQ(binary.BigEndian.AppendUint64(nil, uint64(dest)))).
+		FilterBytes(8,
+			query.GTE(binary.BigEndian.AppendUint64(nil, uint64(seqNumRange.Start()))),
+			query.LTE(binary.BigEndian.AppendUint64(nil, uint64(seqNumRange.End()))),
+		).
+		OrderBy(query.SortByTxLT, query.ASC).
+		Limit(int(seqNumRange.End()-seqNumRange.Start()+1)). //nolint:gosec // conversion is safe in this context
+		Execute(ctx, a.logPoller.GetStore())
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to query onRamp logs: %w", err)
+	}
+	a.lggr.Infow("queried messages between sequence numbers",
+		"numMsgs", len(res.Logs),
+		"sourceChainSelector", a.chainSelector,
+		"seqNumRange", seqNumRange.String(),
+	)
+
+	msgs := make([]ccipocr3.Message, 0)
+	for _, log := range res.Logs {
+		// convert event to generic CCIP event
+		event, err := a.convertCCIPMessageSent(
+			&log.TypedData,
+		)
+		if err != nil {
+			a.lggr.Errorw("failed to convert event", "err", err, "log", log)
+			continue
+		}
+
+		// validate event
+		if err := chainaccessor.ValidateSendRequestedEvent(event, a.chainSelector, dest, seqNumRange); err != nil {
+			a.lggr.Errorw("validate send requested event", "err", err, "message", event)
+			continue
+		}
+		event.Message.Header.OnRamp = ccipocr3.UnknownAddress(onrampAddr.String())
+		event.Message.Header.TxHash = string(log.TxHash[:]) // TODO: add LT?
+		msgs = append(msgs, event.Message)
+	}
+
+	msgsWithoutDataField := make([]ccipocr3.Message, len(msgs))
+	for i, msg := range msgs {
+		msgsWithoutDataField[i] = msg.CopyWithoutData()
+	}
+
+	a.lggr.Debugw("decoded messages between sequence numbers",
+		"msgsWithoutDataField", msgsWithoutDataField,
+		"sourceChainSelector", a.chainSelector,
+		"seqNumRange", seqNumRange.String(),
+	)
+
+	// Create message ID strings for logging
+	msgIDs := make([]string, len(msgsWithoutDataField))
+	for i, m := range msgsWithoutDataField {
+		msgIDs[i] = fmt.Sprintf("%d.%d", m.Header.SequenceNumber, m.Header.MessageID)
+	}
+	a.lggr.Infow("decoded message IDs between sequence numbers",
+		"seqNum.MsgID", msgIDs,
+		"sourceChainSelector", a.chainSelector,
+		"seqNumRange", seqNumRange.String(),
+	)
```

Comment:
This looks like general code that is in no way ton specific and could be moved up to the chainaccessor framework. We should just create return messages, and they should auto-magically get logged as such.